src/marlin/types.ts

```diff
@@ -0,0 +1,188 @@
+import { Event } from 'ethers';
+import { ISubscribeOptions } from '../interfaces';
+import { Comp } from './contractTypes/Comp';
+import { GovernorAlpha } from './contractTypes/GovernorAlpha';
+import { Timelock } from './contractTypes/Timelock';
+
+// Used to unwrap promises returned by contract functions
+// TODO: What functions do I need to UnPromisify?
+type UnPromisify<T> = T extends Promise<infer U> ? U : T;
+// export type CompSOMETHING= UnPromisify<ReturnType<Comp['functions'][]>>
+
+// API is imported contracts classes
+export type Api = Comp | GovernorAlpha | Timelock;
```

Comment:
Why a union type? Shouldn't the API contain all of them, not just one (the Moloch contract is a union because the API is _either_ a type 1 _or_ a type 2 contract, with both containing all requisite functionality)? In this case, I'd synthesize an API as a little interface containing all three contracts.